From 34eed5fdd431a040b3982d2800bfc3897802fce5 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Wed, 30 Nov 2022 10:21:21 +0100
Subject: [PATCH 12/13] delta-ag7648: make sure i2c-i801 is loaded before
 creating devices

Make sure i2c-i801 is loaded before we create devices so that the
pca954x mux will use busses 2-9 and not claim the one of i2c-i801.

Since the mux is attached to i2c-ismt, we do not need request it, as
it will always get a bus assigned before we can even register the
mux device.

Upstream-Status: Inactive-Upstream [lastcommit: 2024-09-21]
Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 .../builds/x86-64-delta-ag7648-i2c-mux-setting.c       | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-i2c-mux-setting.c b/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-i2c-mux-setting.c
index 206094fbc5a1..cedfa4e3da62 100644
--- a/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-i2c-mux-setting.c
+++ b/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-i2c-mux-setting.c
@@ -517,6 +517,16 @@ static int __init delta_switch_init(void)
 
 	pr_debug("DMI Matched %s\n", dmi_id->ident);
 
+	/*
+	 * Make sure i2c-i801 is loaded before we create devices so that the
+	 * pca954x mux will use busses 2-9 and not claim the one of i2c-i801.
+	 *
+	 * Since the mux is attached to i2c-ismt, we do not need request it, as
+	 * it will always get a bus assigned before we can even register the
+	 * mux device.
+	 */
+	request_module("i2c-i801");
+
 	dta_switch = delta_switch_prepare((void *)dmi_id->driver_data);
 	if (IS_ERR(dta_switch))
 		return PTR_ERR(dta_switch);
-- 
2.42.0

